src/program.cpp

```diff
@@ -189,559 +458,56 @@ bool spir_binary::validate() const {
-        }
-
-        if (entry.kind != clspv::version0::DescriptorMapEntry::Kind::KernelArg)
-            return false;
+bool spir_binary::strip_reflection(std::vector<uint32_t>* stripped) {
+    const spvtools::MessageConsumer consumer =
+        [](spv_message_level_t level, const char*,
+           const spv_position_t& position, const char* message) {
 
-        kernel_argument arg;
-        arg.name = entry.kernel_arg_data.arg_name;
-        arg.pos = entry.kernel_arg_data.arg_ordinal;
-        if (entry.kernel_arg_data.arg_kind == clspv::ArgKind::Local) {
-            arg.kind = kernel_argument_kind::local;
-            arg.local_elem_size = entry.kernel_arg_data.local_element_size;
-            arg.local_spec_id = entry.kernel_arg_data.local_spec_id;
-        } else {
-            arg.descriptorSet = entry.descriptor_set;
-            arg.binding = entry.binding;
-            arg.offset = entry.kernel_arg_data.pod_offset;
-            switch (entry.kernel_arg_data.arg_kind) {
-            case clspv::ArgKind::Buffer:
-                arg.kind = kernel_argument_kind::buffer;
-                break;
-            case clspv::ArgKind::BufferUBO:
-                arg.kind = kernel_argument_kind::buffer_ubo;
-                break;
-            case clspv::ArgKind::Pod:
-                arg.kind = kernel_argument_kind::pod;
-                break;
-            case clspv::ArgKind::PodUBO:
-                arg.kind = kernel_argument_kind::pod_ubo;
-                break;
-            case clspv::ArgKind::PodPushConstant:
-                arg.kind = kernel_argument_kind::pod_pushconstant;
+#define msgtpl "spvtools says '%s' at position %zu"
+            switch (level) {
+            case SPV_MSG_FATAL:
+            case SPV_MSG_INTERNAL_ERROR:
+            case SPV_MSG_ERROR:
+                cvk_error(msgtpl, message, position.index);
                 break;
-            case clspv::ArgKind::ReadOnlyImage:
-                arg.kind = kernel_argument_kind::ro_image;
+            case SPV_MSG_WARNING:
+                cvk_warn(msgtpl, message, position.index);
                 break;
-            case clspv::ArgKind::WriteOnlyImage:
-                arg.kind = kernel_argument_kind::wo_image;
+            case SPV_MSG_INFO:
+                cvk_info(msgtpl, message, position.index);
                 break;
-            case clspv::ArgKind::Sampler:
-                arg.kind = kernel_argument_kind::sampler;
+            case SPV_MSG_DEBUG:
+                cvk_debug(msgtpl, message, position.index);
                 break;
-            default:
-                return false;
-            }
-            if (arg.is_pod()) {
-                arg.size = entry.kernel_arg_data.pod_arg_size;
             }
-        }
+#undef msgtpl
+        };
 
-        m_dmaps[entry.kernel_arg_data.kernel_name].push_back(arg);
+    spvtools::Optimizer opt(m_target_env);
+    opt.SetMessageConsumer(consumer);
+    opt.RegisterPass(spvtools::CreateStripReflectInfoPass());
```

Comment:
Maybe the pass factory should have had "clspv" in the name to be consistent with the instruction set name.

Reply:
The reflection stripping pass existed before my instruction set and handles non-semantic instructions generally and also the HLSL reflection instructions and decorations.

Reply:
Ok, that's a bit counter-intuitive. I'm wondering whether generic stripping of all non-semantic instructions shouldn't be a separate pass. I'm not familiar enough with the HLSL bits to have an informed opinion though.